### ISSUE TITLE
"Add Shared Project Reference" missing

### ISSUE THREAD

**Issue by Cybrosys** [aspnet/Tooling#426]
How do you add a reference to a .shproj (Shared Project) project, the add reference dialog does not list shared projects?

Update:
I found I could use the "compile" in projects.json to include my shared project's folder for compilation.

**Comment by sayedihashimi** [comment]
@Cybrosys as you stated dnx/dotnet projects have support to share files between projects. We haven't added first class support for this in Visual Studio yet, but it's on our radar. It's not a high priority item though as we have many other features in front of it.

**Comment by cloudcrypt** [comment]
@sayedihashimi, any approximate ETA for this feature?
also, @Cybrosys, could you elaborate a little on how you made that work, thanks!

**Comment by Cybrosys** [comment]
@cloudcrypt of course. Open up your project.json files that you want to have the shared files added to, then add the following in each:
`"compile": [
    "<relative path to folder containing cs files>/*.cs"
  ]`

In my case I have a shared project called Merlot.Dto. The projects that need it have the following in their project.json files:
`"compile": [
    "../Merlot.Dto/*.cs"
  ]`

**Comment by sayedihashimi** [comment]
@cloudcrypt we have not started working on it, so it will not make the RC2 release. I'm hoping that we can get this in the release after that, but I cannot commit to that at this time.

From what I understand this issue is causing some pain, but it's not blocking anything. if there is anything that you feel that you are blocked on please let me know.

**Comment by MV10** [comment]
A combination of this issue and the inability to reference DLLs (frequently reported and seemingly ignored, [for example](https://github.com/aspnet/Home/issues/1612)) is a blocking issue for us. We have several DLLs that must remain in an older, incompatible .NET Framework (for an older client app) so we can't reference the DLL directly. We tried using shared projects to maintain parallel old-framework and core builds, but core-compatible class libraries can't reference shared projects either (not just ASP.NET Core, apparently). Just in case, we then tried adding the shared projects directly to the ASP.NET Core but of course that is also broken.

**Comment by resnyanskiy** [comment]
The issue is fixed for .csproj file.
